nixos/modules/services/misc/home-assistant.nix

```
@@ -13,7 +13,7 @@ let
     ${pkgs.remarshal}/bin/json2yaml -i ${configJSON} -o $out
     # Hack to support secrets, that are encoded as custom yaml objects,
     # https://www.home-assistant.io/docs/configuration/secrets/
-    sed -i -e "s/'\!secret \(.*\)'/\!secret \1/" $out
+    sed -i -e "s/'\!\([^[:space:]]\+\) \(.*\)'/\!\1 \2/" $out
```

I wonder if we should allow to escape `\!` at the beginning of a string:

```suggestion
    sed -i -e "s/'\!\(([^a-z_]\+\) \(.*\)'/\!\1 \2/;s/^\!\!/\!/;" $out
```

untested, but hopefully comprehensible.

---

I understand that as if you want a literal exclamation mark at the beginning of your string use "!!something" => "!something". Is that correct?

---

With this change the string doesn't get unquoted unfortunately.

---

For some reason the `([a-z_]\+)` part does not work for my themes function call. Not sure why.

```
      frontend = {
        themes = "!include_dir_merge_named themes";
      };
```

---

My mistake, probably didn't properly update the nixpkgs checkout when testing.